### ISSUE TITLE
Strange behavior in `odgi inject`

### ISSUE THREAD

**Issue by anshumanmohan** [pangenome/odgi#495]
After discussion with @AndreaGuarracino on Matrix, I understand that `odgi inject` should work as follows:
You have `graph.gfa`:
```
S 1 AA
S 2 CC
S 3 GG
P x 1+,2+,3+ *
```
And `new_paths.bed`:
```
x    0    4    y    
```
Now running `odgi inject` and `odgi view` should give you:
```
S 1 AA
S 2 CC
S 3 GG
P x 1+,2+,3+ *
P y 1+,2+ *
```
There are other interesting things about chopping nodes as necessary, but let's put those aside for a bit.

The problem is that, when I run the above, I get:
```
S 1 AA
S 2 TT
S 3 GC
P x 1+,2+,3-	*
P y 1+,2+	*
```
i.e., the last link of path x is being reversed, and there isn't a reverse-complementing of the relevant segment S 3, so I believe the path's sequence is genuinely being changed.

This behavior remains the same if I _start_ with a reverse traversal over the last segment. For example,
Using `graph_3_rev.gfa`:
```
S 1 AA
S 2 CC
S 3 GG
P x 1+,2+,3- *
```
And the same `new_paths.bed` as before gives me the same output as above, which in this case is correct. 

Though my exploration is by no means exhaustive, I have tried this with a few other versions of reverse-traversals along path x, and with alternate order of node traversals along path x. The result seems to be the same: the last link is converted into a reverse traversal regardless of how it started.

I'd appreciate any thoughts!

**Comment by AndreaGuarracino** [comment]
Hi @anshumanmohan, could you please share all the files you mention (GFA and BED) and the command lines (I suppose odgi inject ... | odgi view ...). I can't reproduce your issue.

**Comment by anshumanmohan** [comment]
Hm, that's very strange. I have checked that I'm the latest commit of master. And also, I just narrowed it down further! `inject` is not the issue, I think there's somehow something wrong with `build`.

File `i.gfa`:
```
H	VN:Z:1.0
S	1	AAAA
S	2	TTTT
S	3	GGGG
S	4	CCCC
P	x	1+,2+,3+,4+ *
L	1	+	2	+	0M
L	2	+	3	+	0M
L	3	+	4	+	0M
```

Comand: `odgi build -g i.gfa -o - | odgi view -g -i - > i_via_odgi.gfa`

Output file `i_via_odgi.gfa`:
```
H	VN:Z:1.0
S	1	AAAA
L	1	+	2	+	0M
S	2	TTTT
L	2	+	3	+	0M
S	3	GGGG
L	3	+	4	+	0M
S	4	CCCC
P	x	1+,2+,3+,4-	*
```


So then I figured I'd check if's `build` or `view`...
Commands: 
`odgi build -g i.gfa -o i.og` (no complaint)
`odgi validate -i i.og`
Output:
`[odgi::validate] error: the path x does not respect the graph topology: the link 3+,4- is missing.` 
This is strange indeed, since the input file (`i.gfa`, above) has no path with link `3+,4-`. I think `build` introduced that, and is now complaining about the missing link.

**Comment by anshumanmohan** [comment]
Please let me know if you can recreate this behavior! Meanwhile I'll also my colleague @susan-garry to please take a look, in case it is something local to us...

**Comment by AndreaGuarracino** [comment]
I know the guilty! I had a similar "discovery" in the past.

The issue is that your input GFA is not 100% correct. In particular, this line:

```
P	x	1+,2+,3+,4+ *
```

has to be

```
P	x	1+,2+,3+,4+	*
```

Note the TAB between 4+ and *, instead of the space.

**Comment by anshumanmohan** [comment]
Aha, the tab strikes again! I think it may be worth changing the behavior of odgi to complain instead of forging ahead in this case. Would it be okay if I did that in a PR, or is that known to be a bad idea?

**Comment by AndreaGuarracino** [comment]
Yes, please. It would be very nice, hoping not to degrade the performance of odgi build with large graphs.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by anshumanmohan** [comment]
Sounds good; feel free to assign it to me if you'd like to make it official! 

Re: performance, it occurs to me that there could just be a command-line flag that a user can pass in that says "please check my GFA is well-formed". If the user does not pass that flag, it'll go faster because we won't run a check per line, but they accept the risk that passing in a badly-formed GFA may lead to quiet undefined behavior.
